libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapLoader.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.maps.MapView
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+internal class MapLoader {
+
+    private val _mapViewAttached: MutableStateFlow<MapView?> = MutableStateFlow(null)
+    val mapViewAttached: StateFlow<MapView?> = _mapViewAttached.asStateFlow()
+    private val _mapViewDetached: MutableStateFlow<MapView?> = MutableStateFlow(null)
+    val mapViewDetached: StateFlow<MapView?> = _mapViewDetached.asStateFlow()
+
+    fun updateMapView(mapView: MapView) {
+        _mapViewDetached.value = mapViewAttached.value
```

Comment:
This flow keeps a reference to the old `MapView`, so it looks like this can be considered a leak. 
Also when we register a listener after `MapView` was updated, `onMapViewDetached` is immediately called with a previous instance, is this expected?

Reply:
Removed this implementation.